### PR TITLE
Bugfix/haystack ai version

### LINKED ISSUE #8: Address security vulnerability with haystack-ai

### PR DESCRIPTION
**Pull request by kuraisle**
haystack 2.1 has a security vulnerability related to the jinja2 templates used to build prompts. Upgrading to 2.3.1 meant I had to fix numpy at 1.* because of breaking changes when using numpy 2.*

### CODE REVIEW

File: Carrot-Assistant/requirements.txt [hunk 1]
```diff
@@ -11,4 +11,4 @@ streamlit==1.36.0
 rapidfuzz==3.5.2
 sqlalchemy==2.0.30
 openssl=3.1.4
```
Comment (Karthi-DStech) [inline Carrot-Assistant/requirements.txt:14]: We had a discussion to remove openssl and it was removed earlier yeah?

Comment (kuraisle) [reply Carrot-Assistant/requirements.txt:14]: Just testing 😛

**Comment by Karthi-DStech** [comment]
There is a conflict to be resolved James. Once we've resolved it, we are good to merge with the main.

**Comment by Karthi-DStech** [comment]
I've solved the conflict and merge it to the main.

**Pull request merged**